tests/unit/vectors.frag

```diff
@@ -23,7 +23,16 @@ float withExtraComponents() {
     vec2 v2 = vec2(1);
     vec3 v3 = vec3(1, 2, v2.x);
     vec4 v4 = vec4(v2, v3.xy);
-    return v2.x + v3.x + v4.x;
+    vec3 v5 = vec3(1, v3.rg);
+    return v2.x + v3.x + v4.x + v5.x;
+}
+
+float withExtraComponentsUnsafe(in vec3 a) {
+    vec3 v1 = vec3(a.x);
+    vec3 v2 = vec3(1.0, a.x);
```

Comment:
> An alternative could be to do the transformation only when there are at least two arguments in the constructor call?

If I understand you, that would apply to calls like this, which are AIUI unsafe to transform (this is equivalent to `vec3(1.0, a.xx)` not `vec3(1.0, a.xy)`).

Reply:
`vec3(1.0, a)` is equivalent to `vec3(1.0, a.xy)`

So I think that `a.xy` can become `a` whenever it's the last argument and there's more than one argument.